src/pkg/packager/lint/lint.go

```diff
@@ -25,120 +22,109 @@ import (
 	"github.com/xeipuuv/gojsonschema"
 )
 
-// ZarfSchema is exported so main.go can embed the schema file
-var ZarfSchema embed.FS
-
-func getSchemaFile() ([]byte, error) {
-	return ZarfSchema.ReadFile("zarf.schema.json")
+// FileLoader is an interface for reading files, it decouples the lint package from the go embed package which enables testing
+type FileLoader interface {
+	ReadFile(path string) ([]byte, error)
 }
 
-// Validate validates a zarf file against the zarf schema, returns *validator with warnings or errors if they exist
-// along with an error if the validation itself failed
-func Validate(ctx context.Context, createOpts types.ZarfCreateOptions) (*Validator, error) {
-	validator := Validator{}
-	var err error
+// ZarfSchema is exported so main.go can embed the schema file
+var ZarfSchema FileLoader
```

Comment:
Why are we not passing the schema as a function argument? I would like to avoid the use of global variables as much as possible.

Reply:
It's global because the value is set in main.go through go embed. Go embed has to happen in main because we can't use file backtrackers in go embed, for example, I can't do "../../../../zarf.schema.json". I think the only other way of passing it through to the commands would be adding it as a value to the context.